jina/peapods/zmq/__init__.py

```diff
@@ -509,14 +514,24 @@ def close(self, flush: bool = True, *args, **kwargs):
         :param args: Extra positional arguments
         :param kwargs: Extra key-value arguments
         """
-        if not self.is_closed and self.in_sock_type == zmq.DEALER:
+
+        # if Address already in use `self.in_sock_type` is not set
+        if (
+            not self.is_closed
+            and hasattr(self, 'in_sock_type')
```

Comment:
This can be removed again. I fixed the resulting error in #2853 via making sure, this is always set to `None` in the first place.

Reply:
will remove in follow-up PR if that's okey